projects/gateway2/proxy_syncer/proxy_syncer.go

```diff
@@ -75,109 +103,531 @@ var kubeGatewayProxyLabels = map[string]string{
+	configMapClient := kclient.New[*corev1.ConfigMap](s.istioClient)
+	configMaps := krt.WrapClient(configMapClient, krt.WithName("ConfigMaps"))
+
+	secretClient := kclient.New[*corev1.Secret](s.istioClient)
+	secrets := krt.WrapClient(secretClient, krt.WithName("Secrets"))
+
+	authConfigs := setupCollectionDynamic[extauthkubev1.AuthConfig](
+		ctx,
+		s.istioClient,
+		extauthkubev1.SchemeGroupVersion.WithResource("authconfigs"),
+		krt.WithName("KubeAuthConfigs"),
+	)
 
-	var (
-		secretsWarmed bool
-		// totalResyncs is used to track the number of times the proxy syncer has been triggered
-		totalResyncs int
+	rlConfigs := setupCollectionDynamic[rlkubev1a1.RateLimitConfig](
+		ctx,
+		s.istioClient,
+		rlkubev1a1.SchemeGroupVersion.WithResource("ratelimitconfigs"),
+		krt.WithName("KubeRateLimitConfigs"),
 	)
-	resyncProxies := func() {
-		if !secretsWarmed {
-			return
-		}
-		totalResyncs++
-		contextutils.LoggerFrom(ctx).Debugf("resyncing k8s gateway proxies [%v]", totalResyncs)
-		stopwatch := statsutils.NewTranslatorStopWatch("ProxySyncer")
-		stopwatch.Start()
-		var (
-			proxies gloo_solo_io.ProxyList
-		)
-		defer func() {
-			duration := stopwatch.Stop(ctx)
-			contextutils.LoggerFrom(ctx).Debugf("translated and wrote %d proxies in %s", len(proxies), duration.String())
-		}()
 
-		var gwl gwv1.GatewayList
-		err := s.mgr.GetClient().List(ctx, &gwl)
-		if err != nil {
-			// This should never happen, try again?
-			return
+	kubeUpstreams := setupCollectionDynamic[glookubev1.Upstream](
+		ctx,
+		s.istioClient,
+		glookubev1.SchemeGroupVersion.WithResource("upstreams"),
+		krt.WithName("KubeUpstreams"),
+	)
+
+	// helper collection to map from the kube Upstream type to the gloov1.Upstream wrapper
+	glooUpstreams := krt.NewCollection(kubeUpstreams, func(kctx krt.HandlerContext, u *glookubev1.Upstream) *upstream {
+		// TODO: not cloning, this is already a copy from the underlying cache, right?!
+		glooUs := &u.Spec
+		glooUs.Metadata = &core.Metadata{}
+		glooUs.GetMetadata().Name = u.GetName()
+		glooUs.GetMetadata().Namespace = u.GetNamespace()
+		us := &upstream{glooUs}
+		return us
+	}, krt.WithName("GlooUpstreams"))
+
+	serviceClient := kclient.New[*corev1.Service](s.istioClient)
+	services := krt.WrapClient(serviceClient, krt.WithName("Services"))
+
+	inMemUpstreams := krt.NewManyCollection(services, func(kctx krt.HandlerContext, svc *corev1.Service) []upstream {
+		uss := []upstream{}
+		for _, port := range svc.Spec.Ports {
+			us := kubeupstreams.ServiceToUpstream(ctx, svc, port)
+			uss = append(uss, upstream{us})
 		}
+		return uss
+	}, krt.WithName("InMemoryUpstreams"))
+
+	finalUpstreams := krt.JoinCollection([]krt.Collection[upstream]{glooUpstreams, inMemUpstreams})
 
-		pluginRegistry := s.k8sGwExtensions.CreatePluginRegistry(ctx)
-		rm := reports.NewReportMap()
-		r := reports.NewReporter(&rm)
+	podClient := kclient.New[*corev1.Pod](s.istioClient)
```

Comment:
```suggestion
	podClient := kclient.NewFiltered[*corev1.Pod](s.istioClient, kclient.Filter{
		ObjectTransform: kube.StripPodUnusedFields,
	})
```

This gets rid of ManagedFields which is large and almost definitely unused